### PR TITLE
HomeWork jv-lottery

### PR DESCRIPTION
**Pull request by ViktoriiaSvidovska**

### CODE REVIEW

**Review by Rommelua — CHANGES_REQUESTED** [review]
Java Cl checks failed. Before making a Pull Request (or a new commit to an already existing PR), you need to write the mvn clean package command in the terminal and make sure that all checks pass. After that, the mentor will be able to review the homework

image

File: src/main/java/core/basesyntax/Lottery.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int maxNumber = 100;
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Lottery.java:6]:
```suggestion
    private static final int MAX_NUMBER = 100;
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+    public static Ball getRandomBall() {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Lottery.java:8]: image

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        Colors color = ColorSupplier.getRandomColor();
+        int number = new Random().nextInt(maxNumber);
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Lottery.java:10]: move random to the class field

File: src/main/java/core/basesyntax/Application.java [hunk 4]
```diff
@@ -2,6 +2,12 @@
...
 public class Application {
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+        Ball ball1 = Lottery.getRandomBall();
+        Ball ball2 = Lottery.getRandomBall();
+        Ball ball3 = Lottery.getRandomBall();
+
+        System.out.println(ball1);
+        System.out.println(ball2);
+        System.out.println(ball3);
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Application.java:11]:
```suggestion
        Ball ball1 = Lottery.getRandomBall();
        Ball ball2 = Lottery.getRandomBall();
        Ball ball3 = Lottery.getRandomBall();

        System.out.println(ball1);
        System.out.println(ball2);
        System.out.println(ball3);
```

use loop for this

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Application.java [hunk 5]
```diff
@@ -2,6 +2,16 @@
 
 public class Application {
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+        int numberOfBalls = 3;
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:5]: either hardcode it in the for loop or make it a class constant

File: src/main/java/core/basesyntax/Application.java [hunk 6]
```diff
@@ -2,6 +2,16 @@
...
+
+        for (int i = 0; i < numberOfBalls; i++) {
+            Lottery lottery = new Lottery();
+            balls[i] = lottery.getRandomBall();
+        }
+
+        for (Ball ball : balls) {
+            System.out.println(ball);
+        }
     }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:16]:
```suggestion
        Lottery lottery = new Lottery();
        for (int i = 0; i < 3; i++) {
           System.out.println(lottery.getRandomBall());
        }
}
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 7]
```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    private static final Random random = new Random();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]:
```suggestion
    private final Random random = new Random();
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 8]
```diff
@@ -1,7 +1,12 @@
...
+
+    public static Color getRandomColor() {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/ColorSupplier.java:8]:
```suggestion
    public Color getRandomColor() {
```
as was mentioned in the checklist, avoid static methods

File: src/main/java/core/basesyntax/Lottery.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int MAX_NUMBER = 100;
+
+    public Ball getRandomBall() {
+        Color color = ColorSupplier.getRandomColor();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Lottery.java:9]: Random, ColorSupplier should be class fields

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by uliana-andrushchuk — APPROVED** [review]

**Pull request closed without merging**